agentlightning/runner/agent.py

```diff
@@ -277,58 +278,63 @@ async def _post_process_rollout_result(
         store = self.get_store()
 
         trace_spans: list[ReadableSpan] | list[Span] = []
-        result_recognized: bool = False
 
         # Case 0: result is None
         if raw_result is None:
             trace_spans = self._tracer.get_last_trace()
-            result_recognized = True
 
         # Case 1: result is a float (final reward)
-        if isinstance(raw_result, (bool, int, float)):
-            if isinstance(raw_result, (bool, int)):
-                logger.warning(
-                    f"{self._log_prefix(rollout.rollout_id)} Reward is not a number, got: {type(raw_result)}. "
-                    "Auto converting to float."
-                )
-                raw_result = float(raw_result)
+        if isinstance(raw_result, float):
             # Preserve the existing spans before another span is emitted
             trace_spans = list(self._tracer.get_last_trace())
             # This will NOT emit another span to the tracer
             reward_span = emit_reward(raw_result, propagate=False)
             # We add it to the store manually
-            await store.add_otel_span(rollout.rollout_id, rollout.attempt.attempt_id, reward_span)
+            try:
+                await store.add_otel_span(rollout.rollout_id, rollout.attempt.attempt_id, reward_span)
+            except (ServerShutdownError, asyncio.CancelledError):
+                # Server is shutting down or request was cancelled - handle gracefully without traceback
+                logger.debug(
```

Comment:
I think this isn't the only place with such issue. We'd better handle it in store client.